visualiser/index.py

```diff
@@ -0,0 +1,102 @@
+from pprint import pprint
+import matplotlib.pyplot as plt
+import pandas as pd
+import sys
+
+from rich.console import Console
+console = Console(record = True)
+
+args = sys.argv[1:]
+if len(args) > 1:
+    console.print("Too many arguments passed in.", style="bold red")
+    sys.exit()
+elif len(args) == 1:
+    op_id = int(args[0])
+
+def visualise_op(op_id, op_df, msg_df):
+    fig, ax = plt.subplots(figsize=(30, 30))
+    
+    for _, row in op_df.iterrows():
+        if int(row["id"]) == int(op_id):
+            op_src = row["src"]
+
+    try:
+        op_src
+    except NameError:
+        console.print("Couldn't find an operation with the specified ID: " + str(op_id), style="bold red")
+        sys.exit()
+
+    ax.set_ylim(ymin=op_id * 0.8, ymax= op_id * 1.2)
```

Comment:
Why is the `ylim` set based on `op_id`?

Reply:
So that we can "zoom in" the y-axis to that horizontal line. It might be unnecessary - I am yet to find out.